### PR TITLE
[terra-menu] - Fixed focus highlight issue for menu button

### PR DESCRIPTION
**Pull request by PK106552**
### Summary
Fixes focus not going back to menu button after selecting an item in menu item.

**What was changed:**
Focus styles are added for menu button.

**Why it was changed:**
In Safari the focus is not consistent so to fix this inconsistency of focus style i have added focus styles

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9664 

---

Thank you for contributing to Terra.
@cerner/terra

**Before fix**

https://github.com/cerner/terra-framework/assets/127083404/5459b1f8-6e44-48eb-993c-812d240f16d3

**After fix**

https://github.com/cerner/terra-framework/assets/127083404/98b1b15d-9707-450c-a873-182650bd47cf

### CODE REVIEW

File: packages/terra-framework-docs/src/terra-dev-site/doc/menu/example/BasicMenu.module.scss [hunk 1]
```diff
@@ -1,5 +1,11 @@
 :local {
   .menu-wrapper {
     display: inline-block;
+
+    button {
+      &:focus {
+        box-shadow: 0 0 1px 3px rgba(76, 178, 233, 0.5), 0 0 7px 4px rgba(76, 178, 233, 0.35);
```
Comment (saket2403) [inline packages/terra-framework-docs/src/terra-dev-site/doc/menu/example/BasicMenu.module.scss:7]: Please add the equivalent styles for clinical and fusion themes

Comment (PK106552) [reply packages/terra-framework-docs/src/terra-dev-site/doc/menu/example/BasicMenu.module.scss]: Added focus styles for clinical and fusion themes

**Review by saket2403 — APPROVED** [review]

File: packages/terra-framework-docs/CHANGELOG.md
Comment (sdadn) [inline packages/terra-framework-docs/CHANGELOG.md:1]: +1 nice description of change + fix

**Review by sdadn — COMMENTED** [review]
Is this only a visual change or also a behavioral change? Either way, there should be wdio updates. If there are no screenshot updates, then we should write a wdio test for this scenario.

**Comment by supreethmr** [comment]
> Is this only a visual change or also a behavioral change? Either way, there should be wdio updates. If there are no screenshot updates, then we should write a wdio test for this scenario.

Here we are trying to just update the examples to handle focus styles for anchor button which triggers the menu. In doc-site examples terra-button has been used as anchor buttons for terra-menu. there was observation from Mpage Fusion passthrough where they were not able to get the focus back on terra-button on closing the menu with item selection. 
Since terra-button provides `onFocus()` callback suggestion was to use this callback to handle required focus styles on fusion side instead of making focus style updates on terra-button. Since our doc-site examples also had similar observation of missing focus highlight style on selection of menu-item we updated doc-site examples to provide example for fusion team to implement same on fusion end.

Since we are not making src code change do we still need WDIO test case.

**Review by supreethmr — APPROVED** [review]

File: packages/terra-framework-docs/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed
+  * Fixed issue of focus not going back to menu button after selecting an item in menu item.
```
Comment (supreethmr) [inline packages/terra-framework-docs/CHANGELOG.md:6]:
```suggestion
* Changed
  * Updated `terra-menu` examples with focus highlight styles for terra-button on closing menu with selection of menu-item.
```

Comment (PK106552) [reply packages/terra-framework-docs/CHANGELOG.md]: Updated

**Pull request merged**